paperqa/agents/__init__.py

```diff
@@ -116,15 +116,17 @@ def search_query(
 
 
 def build_index(
-    index_name: str,
-    directory: str | os.PathLike,
-    settings: Settings,
+    index_name: str | None = None,
+    directory: str | os.PathLike | None = None,
+    settings: Settings | None = None,
 ) -> SearchIndex:
     """Build a PaperQA search index, this will also happen automatically upon using `ask`."""
+    settings = get_settings(settings)
     if index_name == "default":
```

Comment:
should this check for `None` too?

Reply:
I was letting `None` be handled by the wrapped function `get_directory_index`. I can do this better to make it clear

Reply:
Just pushed a change to DRY it up with `get_directory_index`